crates/bevy_render/src/render_resource/resource_macros.rs

```diff
@@ -0,0 +1,84 @@
+#[cfg(debug_assertions)]
+#[macro_export]
+macro_rules! render_resource_type {
+    ($wgpu_type:ty) => {
+        Arc<Box<()>>
```

Comment:
Would it be possible to have the type erase value directly in the Arc without extra Box? Also can you please introduce a unit struct representing an erased type instead of using ()? Or even better add an Arc wrapper that prevents access without unerasing the type.

Reply:
i *believe* the box is required to ensure that transmuting is safe. we can be sure that the memory layout of Box<T> is the same as Box<U>, but can't be sure that Arc<T> is the same as Arc<U> due to alignment.

i will add an arc wrapper type, good idea.